src/Microsoft.AspNet.OData.Shared/EdmChangedObjectCollection.cs

```diff
@@ -57,5 +69,288 @@ private void Initialize(IEdmEntityType entityType)
             _edmType = new EdmDeltaCollectionType(new EdmEntityTypeReference(_entityType, isNullable: true));
             _edmTypeReference = new EdmCollectionTypeReference(_edmType);
         }
+
+        /// <summary>
+        /// Patch for Types without underlying CLR types
+        /// </summary>
+        /// <param name="originalCollection"></param>
+        /// <returns>ChangedObjectCollection response</returns>
+        public EdmChangedObjectCollection Patch(ICollection<EdmStructuredObject> originalCollection)
+        {            
+            PatchHandler = new DefaultTypelessPatchHandler(originalCollection, _entityType);
+
+            return CopyChangedValues();
+        }
+
+        /// <summary>
+        /// Handler for users Create, Get and Delete Methods
+        /// </summary>
+        internal TypelessPatchMethodHandler PatchHandler { get; set; }
+
+        /// <summary>
+        /// Patch for EdmChangedObjectCollection, a collection for IEdmChangedObject 
+        /// </summary>
+        /// <returns>ChangedObjectCollection response</returns>
+        public EdmChangedObjectCollection Patch(TypelessPatchMethodHandler patchHandler)
+        {
+            PatchHandler = patchHandler;
+            return CopyChangedValues();
+        }
+
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1800:DoNotCastUnnecessarily")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Maintainability", "CA1506:AvoidExcessiveClassCoupling")]
+        internal EdmChangedObjectCollection CopyChangedValues()
+        {
+            EdmChangedObjectCollection changedObjectCollection = new EdmChangedObjectCollection(_entityType);
+            List<IEdmStructuralProperty> keys = _entityType.Key().ToList();
+
+            foreach (IEdmChangedObject changedObj in Items)
+            {                
+                DataModificationOperationKind operation = DataModificationOperationKind.Update;
+                EdmStructuredObject originalObj = null;
+                string errorMessage = string.Empty;
+                string geterrorMessage = string.Empty;
+                IDictionary<string, object> keyValues = GetKeyValues(keys, changedObj);
+
+                try
+                {
+                    EdmStructuredObject original = null;
+                    EdmDeltaDeletedEntityObject deletedObj = changedObj as EdmDeltaDeletedEntityObject;
+
+                    if (deletedObj != null)
+                    {
+                        operation = DataModificationOperationKind.Delete;
+
+                        if (PatchHandler.TryDelete(keyValues, out errorMessage) != PatchStatus.Success)
```

Comment:
>TryDelete [](start = 41, length = 9)

should we differentiate between delete and remove (i.e., based on reason)?  Remove from a collection would mean it's no longer part of the collection, but might still exist.  Delete would be a hard delete.

Reply:
I believe Here we can safely assume user would either remove it from collection or totally delete it based on their need and the logic can be there in their TryDelete method. For a contained collection I beleieve both are same